backend/de.metas.contracts/src/main/java/de/metas/contracts/modular/log/LogsRecomputationService.java

```diff
@@ -82,6 +92,86 @@ public class LogsRecomputationService
 	@NonNull
 	private final ModularContractLogDAO modularContractLogDAO;
 
+	public void recomputeAllFinalInvoiceRelatedLogsLinkedTo(@NonNull final FlatrateTermId flatrateTermId)
+	{
+		final I_C_Flatrate_Term flatrateTerm = flatrateDAO.getById(flatrateTermId);
+
+		final OrderId flatrateTermOrderId = OrderId.ofRepoId(flatrateTerm.getC_Order_Term_ID());
+		inOutDAO.retrieveByQuery(InOutQuery.builder()
+										 .orderId(flatrateTermOrderId)
+										 .build())
+						.forEach(this::recomputeForInOut);
+
+		if(flatrateTerm.isSOTrx())
+		{
+			shippingNotificationService.getByQuery(ShippingNotificationQuery.builder()
+														   .orderId(flatrateTermOrderId)
+														   .build())
+					.forEach(shippingNotification -> recomputeForShippingNotificationLines(shippingNotification.getLines()));
+		}
+		else
+		{
+			final Set<OrderId> salesOrderIds = orderDAO.streamOrderLines(OrderLineQuery.builder().modularPurchaseContractId(flatrateTermId).build())
+					.map(I_C_OrderLine::getC_Order_ID)
+					.map(OrderId::ofRepoId)
+					.collect(Collectors.toSet());
+
+			inOutDAO.retrieveByQuery(InOutQuery.builder()
+											 .orderIds(salesOrderIds)
+											 .build())
+					.forEach(this::recomputeForInOut);
+
+			shippingNotificationService.getByQuery(ShippingNotificationQuery.builder()
+														   .orderIds(salesOrderIds)
+														   .build())
+					.forEach(shippingNotification -> recomputeForShippingNotificationLines(shippingNotification.getLines()));
+
+			ppOrderDAO.streamManufacturingOrders(ManufacturingOrderQuery.builder()
+														 .modularFlatrateTermId(flatrateTermId)
+														 .build())
+					.map(I_PP_Order::getPP_Order_ID)
+					.map(PPOrderId::ofRepoId)
+					.forEach(this::recomputeForPPOrder);
```

Comment:
I think we should also recompute this log ``de.metas.contracts.modular.computing.purchasecontract.addedvalue.interim.PurchaseModularContractLog``

Reply:
This always exists before final invoice, so no need to recompute it

Reply:
I am thinking about this de.metas.contracts.modular.computing.purchasecontract.addedvalue.interim.AVInterimComputingMethod#splitLogsIfNeeded but fine by me to leave it out for now and fix it later when we test/repair interests